### PR TITLE
Update legacy-release_sbom-generator.yaml

### LINKED ISSUE #29443: SBOM generation to go along w/ every release
### Parent Issue

_No response_

### Task

We need to automate the generation of Software Bill of Materials (SBOM) using GitHub Actions. This automation should ensure that a new SBOM is generated with each new build and release. The preferred SBOM formats are SPDX and CycloneDX.

Additional Information:
- [SPDX Specification](https://spdx.dev/specifications/)
- [CycloneDX Specification](https://cyclonedx.org/specification/overview/)

### Proposed Objective

Security & Privacy

### Proposed Priority

Priority 2 - Important

### Acceptance Criteria

1. A GitHub Actions workflow is created and triggers on every push to the main branch OR on every release.
2. SBOMs in SPDX and CycloneDX formats are generated and stored in the repository.
3. SBOM files are attached to the release artifacts and indicated hyperlink in the changelog site.
4. The process is documented and easy to understand. Change log should have a hyperlink to the latest and greatest SBOM.

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_
```[tasklist]
### Tasks
- [ ] Create GitHub Actions Workflow
- [ ] Integrate SBOM Generation Tool
- [ ] Store and Attach SBOM
- [ ] Testing and Validation
- [ ] Documentation
```

### PR DESCRIPTION
**Pull request by rsh1k**
The workflow will generate the SBOM for each release in the release branch instead of core-test-results.

### CODE REVIEW

**Review by mbiuki — APPROVED** [review]

File: .github/workflows/legacy-release_sbom-generator.yaml [hunk 1]
```diff
@@ -51,12 +51,12 @@ jobs:
...
       - name: Scan Docker Image with Syft
         run: |
-          pipx run anchore_syft dotcms/dotcms:${{ env.DOTCMS_VERSION }} -o cyclonedx-xml > core-test-results/sbom/cyclonedx.json
+          pipx run anchore_syft dotcms/dotcms:${{ env.DOTCMS_VERSION }} -o cyclonedx-xml > core/sbom/cyclonedx.json
 
       - name: Rename SBOM file with dotCMS version
         run: |
-          mkdir -p core-test-results/sbom
-          mv core-test-results/sbom/cyclonedx.json core-test-results/sbom/dotcms-${{ env.DOTCMS_VERSION }}.json
+          mkdir -p core/sbom
```
Comment (mbiuki) [inline .github/workflows/legacy-release_sbom-generator.yaml:58]: separate these steps

**Review by mbiuki — APPROVED** [review]

File: .github/workflows/legacy-release_sbom-generator.yaml [hunk 2]
```diff
@@ -65,7 +68,7 @@ jobs:
 
       - name: Commit and push results to core-test-results repository
         run: |
-          cd core-test-results
+          cd core
           git add sbom/dotcms-${{ env.DOTCMS_VERSION }}.json
           git commit -m "Add SBOM for dotCMS version ${{ env.DOTCMS_VERSION }}" || echo "No changes to commit"
           git push origin master
```
Comment (mbiuki) [inline .github/workflows/legacy-release_sbom-generator.yaml:74]: env:
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}

File: .github/workflows/legacy-release_sbom-generator.yaml [hunk 3]
```diff
@@ -33,41 +35,67 @@ jobs:
...
       - name: Scan Docker Image with Syft
         run: |
-          pipx run anchore_syft dotcms/dotcms:${{ env.DOTCMS_VERSION }} -o cyclonedx-xml > core-test-results/sbom/cyclonedx.json
+          pipx run anchore_syft dotcms/dotcms:${{ env.DOTCMS_VERSION }} -o cyclonedx-xml > core/cyclonedx.json
 
       - name: Rename SBOM file with dotCMS version
         run: |
-          mkdir -p core-test-results/sbom
-          mv core-test-results/sbom/cyclonedx.json core-test-results/sbom/dotcms-${{ env.DOTCMS_VERSION }}.json
+          mv core/cyclonedx.json core/dotcms-${{ env.DOTCMS_VERSION }}.json
```
Comment (mbiuki) [inline .github/workflows/legacy-release_sbom-generator.yaml:70]: maybe name is `sbom_cyclonedx.json`

**Review by mbiuki — APPROVED** [review]

**Comment by spbolton** [comment]
My only concern is that we are adding the commit After the commit we fix the release version and publish to github,   therefore anyone actually looking at the files in the release commit will not see the sbom.     Of course there is a chicken-egg problem here as the workflow is triggered off the release generation so the only way I see of fixing this would be to modify the main release workflow to add the steps before the release commit is created.   It probably would be better to have the sbom combined with the single release commit, but a commit before would also be fine.
If 
@bryanboza 
 is ok with this extra post release commit just to get the sbom in there in the short term, then I am ok with that.   Otherwise to handle this before our release workflow refactor would be to modify the current release workflow to include the required steps rather than its own workflow as we currently have.

File: .github/workflows/legacy-release_sbom-generator.yaml [hunk 4]
```diff
@@ -33,41 +35,64 @@ jobs:
             # Use the input provided in manual run
             latest_tag=${{ github.event.inputs.dotcms_version }}
           fi
+          # Format the tag name: remove 'v' prefix
+          formatted_tag=$(echo "$latest_tag" | sed -e 's/^v//' -e 's/^dotcms-cli-//')
+          # Construct the branch name based on the formatted version
+          branch_name="release-${formatted_tag}"
+          echo "DOTCMS_VERSION=${formatted_tag}" >> $GITHUB_ENV
+          echo "BRANCH_NAME=${branch_name}" >> $GITHUB_ENV
```
Comment (sfreudenthaler) [inline .github/workflows/legacy-release_sbom-generator.yaml:43]: For consideration: 
consider making this a standalone `set dotCMS release version` step.  Or adjust the name & id of the existing combined step to be clear that it's a get&set step.

**Review by sfreudenthaler — APPROVED** [review]
@spbolton left a valid comment.  I'm ok either way we go.  

I do like that we're storing as a file in the root of the repo.  Seems like a good pattern to have it by other repo metadata stuff like the readme.  Appreciate that we're not creating the sbom file with the version in the filename and leveraging github versioning as well.  It'll make future usage/programmatic consumption of the sbom much nicer to work with.

**Review by sfreudenthaler — APPROVED** [review]

**Pull request closed without merging**